source/slang/slang-emit-spirv.cpp

```diff
@@ -7022,6 +7022,15 @@ struct SPIRVEmitContext : public SourceEmitterBase, public SPIRVEmitSharedContex
         if (destPtrType->hasAddressSpace())
             addrSpace = destPtrType->getAddressSpace();
         auto ptrElementType = builder.getPtrType(kIROp_PtrType, sourceElementType, addrSpace);
+
```

Comment:
This code is telling me that the IRSwizzledStore is wrong in the first place, and the root cause is to find out why we generated this faulty inst in some previous stage.

Patching up this in the last minute right before emitting SPIRV is more of a band-aid, and we should figure out why there is type mismatch in the IRSwizzledStore.